### PR TITLE
VIH-6922 connection service

### PR DESCRIPTION
**Pull request by phillipjchambers**
### JIRA link (if applicable) ###

VIH-6922

### Change description ###

Add a connection service which periodically tests that the user can make web requests.

- add ConnectionService service, injected in root
- subscribe to connection status in app component & redirect to ErrorComponent on connection error
- modify ErrorComponent wording as per copy provided (see ticket)
- display the correct error for the current connection status
- update HeaderComponent to hide navigation when there is not connection
- update all existing tests that now depend on ConnectionService
- add new tests for AppComponent & ConnectionService

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/app.module.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export function getSettings(configService: ConfigService) {
         ErrorService,
         PageTrackerService,
         AppInsightsLogger,
-        WindowRef
+        WindowRef,
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/app.module.ts:97]: You might want to check you're using the same linting or prettier rules. I didn't expect this to change.

File: AdminWebsite/AdminWebsite/ClientApp/src/app/error/error.component.html [hunk 2]
```diff
@@ -1,6 +1,31 @@
 <div class="govuk-grid-row">
-  <div class="govuk-grid-column-full">
-    <h1 class="govuk-heading-xl">Sorry, there is a problem with the service</h1>
-    <p class="govuk-body">Please try again later.</p>
-  </div>
+  <ng-container *ngIf="hasConnection$ | async">
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/error/error.component.html:2]: Could `*ngIf="hasConnection$ | async;noConnection"` work here?

**Review by shaed-parkar — APPROVED** [review]
Nice work! I really like how you've implemented the connection service.

File: AdminWebsite/AdminWebsite/ClientApp/src/app/error/error.component.html (hunk 2 above)
Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/error/error.component.html:2]: Using a ngTemplate? Probably, yeah - you might need to use parens, though, like this `*ngIf="(hasConnection$ | async); else noConnection"`

File: AdminWebsite/AdminWebsite/ClientApp/src/app/app.module.ts (hunk 1 above)
Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/app.module.ts:97]: Good point, will remove

**Review by shaed-parkar — APPROVED** [review]

**Review by shaed-parkar — APPROVED** [review]

**Pull request merged**